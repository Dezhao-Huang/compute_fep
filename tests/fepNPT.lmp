# created by fftool

units real
boundary p p p

atom_style full
bond_style harmonic
angle_style harmonic
special_bonds lj/coul 0.0 0.0 0.5

read_data data.eq.lmp
# read_restart restart.*.lmp
# reset_timestep 0

pair_style hybrid lj/cut/coul/long 10.0 10.0
pair_modify tail yes
kspace_style pppm 1.0e-4

pair_coeff    1    1  lj/cut/coul/long    0.0028   2.7800  # Ne Ne
pair_coeff    1    2  lj/cut/coul/long    0.0000   1.0000  # Ne Hw
pair_coeff    1    3  lj/cut/coul/long    0.0208   2.9665  # Ne Ow
pair_coeff    2    2  lj/cut/coul/long    0.0000   1.0000  # Hw Hw
pair_coeff    2    3  lj/cut/coul/long    0.0000   1.0000  # Hw Ow
pair_coeff    3    3  lj/cut/coul/long    0.1554   3.1655  # Ow Ow

variable nsteps index 1000
variable nprint equal ${nsteps}/10
variable ndump equal ${nsteps}/100

variable temp equal 300.0
variable press equal 1.0

neighbor 2.0 bin

timestep 1.0

# velocity all create ${temp} 12345

fix fSHAKE all shake 0.0001 20 0 b 1 a 1

#fix fNVT all nvt temp ${temp} ${temp} 100
fix fNPT all npt temp ${temp} ${temp} 100 iso ${press} ${press} 500


# neon to sodium fep

variable lambda equal ramp(0.0,1.0)
variable sig13 equal 3.33*v_lambda+2.78*(1.0-v_lambda)
variable q1 equal v_lambda

variable dlambda equal 0.1
variable dsig13 equal 3.33*v_dlambda-2.78*v_dlambda
variable dq1 equal v_dlambda

variable nadapt equal ${nsteps}/10
variable nrepeat equal ${nadapt}/2

fix fADAPT all adapt/fep ${nadapt} &
    pair lj/cut/coul/long sigma 1 3 v_sig13 &
    atom charge 1 v_q1 &
    after yes

compute cFEP all fep ${temp} &
    pair lj/cut/coul/long sigma 1 3 v_dsig13 &
    atom charge 1 v_dq1

fix fFEP all ave/time 1 ${nrepeat} ${nadapt} c_cFEP[1] c_cFEP[2] c_cFEP[3] file fepNPT.out


# thermo_style multi
thermo_style custom step etotal evdwl ecoul elong temp vol v_lambda v_sig13 v_q1
thermo ${nprint}

# dump dCONF all custom ${ndump} dump.lammpstrj id mol type element x y z ix iy iz
# dump_modify dCONF element Ne H O

# restart ${nrestart} restart.*.lmp

run ${nsteps} post no

# write_restart restart.*.lmp
# write_data data.*.lmp
