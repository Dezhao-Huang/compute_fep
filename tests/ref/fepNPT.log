LAMMPS (29 Aug 2014)
WARNING: OMP_NUM_THREADS environment is not set. (../comm.cpp:88)
  using 1 OpenMP thread(s) per MPI task
  orthogonal box = (-9.36391 -9.36391 -9.36391) to (9.36391 9.36391 9.36391)
  1 by 1 by 1 MPI processor grid
  reading atoms ...
  649 atoms
  reading velocities ...
  649 velocities
  scanning bonds ...
  1 = max bonds/atom
  scanning angles ...
  1 = max angles/atom
  reading bonds ...
  432 bonds
  reading angles ...
  216 angles
  2 = max # of 1-2 neighbors
  1 = max # of 1-3 neighbors
  1 = max # of 1-4 neighbors
  2 = max # of special neighbors
  0 = # of size 2 clusters
  0 = # of size 3 clusters
  0 = # of size 4 clusters
  216 = # of frozen angles
PPPM initialization ...
  G vector (1/distance) = 0.266541
  grid = 10 10 10
  stencil order = 5
  estimated absolute RMS force accuracy = 0.0275322
  estimated relative force accuracy = 8.29123e-05
  using double precision FFTs
  3d grid and FFT values/proc = 4913 1000
FEP settings ...
  temperature = 300.000000
  tail no
  lj/cut/coul/long sigma 1-1 3-3
  1-1 charge
Memory usage per processor = 7.24501 Mbytes
Step TotEng E_vdwl E_coul E_long Temp Volume lambda sig13 q1 
       0   -2010.0137    434.96614    8806.8609   -11627.051    291.37777    6568.4247            0         2.78            0 
      30    -2008.579    452.75845    8802.4739   -11624.174    279.84784    6565.0904          0.1        2.835          0.1 
      60   -2003.5013    457.19249    8785.9346    -11627.85    296.04594    6559.6496          0.2         2.89          0.2 
      90    -2000.312    462.34095    8760.9596   -11626.576     312.9301    6554.2065          0.3        2.945          0.3 
     120   -1996.3427    476.27332     8765.173   -11628.097    303.10254    6553.8575          0.4            3          0.4 
     150   -1994.8153    444.35416    8776.7245   -11627.426    319.58405    6563.7766          0.5        3.055          0.5 
     180   -1993.3372    427.63336    8815.0626   -11626.523    303.24391    6569.7934          0.6         3.11          0.6 
     210   -1990.4712    394.34456    8845.3229   -11627.217    308.36027    6562.4956          0.7        3.165          0.7 
     240   -1987.0787    421.56504    8800.5228   -11625.883    323.61064    6530.2964          0.8         3.22          0.8 
     270   -1984.7902    465.58713    8741.8549   -11625.956    336.81769    6482.9404          0.9        3.275          0.9 
     300   -1988.5342    509.95762    8700.1657   -11620.417    327.52704    6445.5202            1         3.33            1 
Loop time of 3.4405 on 1 procs (1 MPI x 1 OpenMP) for 300 steps with 649 atoms
