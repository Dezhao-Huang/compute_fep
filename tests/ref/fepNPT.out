# Time-averaged data for fix fFEP
# TimeStep c_cFEP[1] c_cFEP[2] c_cFEP[3]
30 1.9917 0.0404554 6566.19
60 0.411138 0.522727 6560.98
90 0.346278 0.634868 6555.45
120 -0.627023 5.48115 6552.95
150 -0.511426 2.6407 6561.18
180 -1.44701 11.802 6569.15
210 -2.55653 74.5836 6565.84
240 -1.4768 12.3219 6539.75
270 -1.49035 17.1877 6493.93
300 -1.17335 8.68426 6452.14
